lib/models/room.js

```diff
@@ -237,43 +316,226 @@ Room.prototype.getAvatarUrl = function(baseUrl, width, height, resizeMethod,
  };
 
 /**
- * Add some events to this room's timeline. Will fire "Room.timeline" for
- * each event added.
+ * Add a new timeline to this room
+ *
+ * @return {module:models/event-timeline~EventTimeline} newly-created timeline
+ */
+Room.prototype.addTimeline = function() {
```

Comment:
Never, but I see no reason to duplicate the `new EventTimeline` call to every caller (with the potential of getting the roomId on that new timeline wrong). Conceptually, timelines belong to rooms; the room should be responsible for making sure new ones are created correctly.

You objected to it being called `createTimeline`, and suggested renaming it...